### PR TITLE
RavenDB-13652: fix getting minIndex bigger than maxIndex in CanHandle…

### PR DESCRIPTION
**Pull request by garayx**
…LogDivergence

### CODE REVIEW

**Comment by arekpalinski** [comment]
I've scheduled another CI build (previous one has failed due to timeout - don't think it's related though)

@karmeli87 Approved by you?

File: src/Raven.Server/Rachis/Follower.cs [hunk 1]
```diff
@@ -887,11 +887,17 @@ private void NegotiateMatchEntryWithLeaderAndApplyEntries(TransactionOperationCo
...
                 // start the binary search again with those boundaries
-                minIndex = Math.Max(minIndex, _engine.GetFirstEntryIndex(context));
+                minIndex = Math.Min(midpointIndex, _engine.GetFirstEntryIndex(context));
                 maxIndex = midpointIndex;
                 midpointIndex = (minIndex + maxIndex) / 2;
                 midpointTerm = _engine.GetTermForKnownExisting(context, midpointIndex);
 
+                if (maxIndex < minIndex)
+                {
+                    Debug.Assert(false, "This is a safeguard against any potential bug here, so in worst case we request the entire snapshot");
```
Comment (karmeli87) [inline src/Raven.Server/Rachis/Follower.cs:903]: Let's add logging here

**Comment by karmeli87** [comment]
Yes

**Review by arekpalinski — APPROVED** [review]

File: src/Raven.Server/Rachis/Follower.cs (hunk 1 above)
Comment (garayx) [reply src/Raven.Server/Rachis/Follower.cs:903]: done

**Pull request merged**